vispy/visuals/polygon.py

```diff
@@ -38,17 +39,40 @@ class PolygonVisual(Visual):
     """
     def __init__(self, pos=None, color='black',
                  border_color=None, border_width=1, **kwargs):
-        super(PolygonVisual, self).__init__(**kwargs)
-
-        self.mesh = MeshVisual()
-        self.border = LineVisual()
+        self._mesh = MeshVisual()
+        self._border = LineVisual()
         self._pos = pos
         self._color = Color(color)
         self._border_width = border_width
         self._border_color = Color(border_color)
+
         self._update()
-        #glopts = kwargs.pop('gl_options', 'translucent')
-        #self.set_gl_options(glopts)
+        CompoundVisual.__init__(self, [self._mesh, self._border], **kwargs)
+        self._mesh.set_gl_state(polygon_offset_fill=True,
```

Comment:
That's a great resource! An important point form that page is that `polygon_offset_fill` does not apply when drawing `GL_LINES`. So you are applying the offset to the mesh, but not the border. However there is a bug here--the AGG line drawing mode uses triangles, so we need to make sure `polygon_offset_fill` is disabled for the line even though it is not necessary with the current tests.

Eventually, we need to decide how best to avoid letting each visual clobber the settings of the others; probably we just need a complete set of GL state variables specified for all visuals (but I worry that may involve a performance hit).